### PR TITLE
Add support for Google Firebase

### PR DESCRIPTION
**Pull request by Draftdog**
The problem with enabling Google Firebase is two-fold:
- there is no way to have the `google-services.json` file processed in a way such as the Gradle plugin is doing 
- Firebase .aar artifacts have `${applicationId}` variables in their `AndroidManifest.xml` resources, these need to be replaced with the application's package name so that it is possible to enable the merging of the manifests, otherwise the build breaks when generating `R.java`

Changes to the plugin are all located in the generate-sources phase. I copied the code from the Gradle plugin and mavenized it. Next I ensured that the firebase .aar artifacts have the ${applicationId} variables filtered.

### CODE REVIEW

**Comment by william-ferguson-au** [comment]
You should really be providing a pull request from a branch, not from master.
Otherwise you risk having extraneous items mixed into the PR.

File: src/main/java/com/simpligility/maven/plugins/android/phase01generatesources/GoogleServicesProcessor.java [hunk 1]
```diff
@@ -0,0 +1,488 @@
...
+import java.io.File;
+import java.io.FileNotFoundException;
+import java.io.IOException;
+import java.util.Map;
+import java.util.TreeMap;
+
+import static com.google.common.base.Charsets.UTF_8;
+
+/**
+ * This code is based on the corresponding Gradle plugin. I have tried to keep the differences to a minimum (apart from
```
Comment (william-ferguson-au) [inline src/main/java/com/simpligility/maven/plugins/android/phase01generatesources/GoogleServicesProcessor.java]: But what does the class do?
A class comment should describe the responsibility of the class.

**Comment by Draftdog** [comment]
Sorry about that. You mean I should have forked one of the existing branches, or do I create my own locally? In case of the former can you tell me which one?

File: src/main/java/com/simpligility/maven/plugins/android/phase01generatesources/GoogleServicesProcessor.java (hunk 1 above)
Comment (Draftdog) [inline src/main/java/com/simpligility/maven/plugins/android/phase01generatesources/GoogleServicesProcessor.java]: Fair enough, I will add some more documentation. Meanwhile, could you guys let me know whether this kind of feature is something you would consider adding (assuming all criteria are met)?

**Comment by william-ferguson-au** [comment]
No, you should create a branch from master and work on that. That way if you have changes for another issue they can be done and submitted in their own branch/PR. Ie don't cross the streams.

I'll leave it for the other committers to comment on relevance of the PR. I haven't come across an instance where this affects me, so I don't feel I understand it.

File: pom.xml [hunk 2]
```diff
@@ -605,6 +605,21 @@
       <version>3.5.1</version>
       <type>jar</type>
     </dependency>
+    <dependency>
+      <groupId>com.google.code.gson</groupId>
+      <artifactId>gson</artifactId>
+      <version>2.8.0</version>
+    </dependency>
+    <dependency>
+      <groupId>com.google.guava</groupId>
```
Comment (mosabua) [inline pom.xml]: There dependencies are WAY too old. They need to be replaced with something newer..

File: README.asciidoc [hunk 3]
```diff
@@ -1,5 +1,16 @@
 == ANDROID MAVEN PLUGIN
 
+This is a fork of the https://github.com/simpligility/android-maven-plugin[Android Maven Plugin].
```
Comment (mosabua) [inline README.asciidoc]: Can the doc be moved into a feature documentation instead .. in src/site...

File: src/main/java/com/simpligility/maven/plugins/android/phase01generatesources/GoogleServicesProcessor.java (hunk 1 above)
Comment (mosabua) [inline src/main/java/com/simpligility/maven/plugins/android/phase01generatesources/GoogleServicesProcessor.java]: I am happy to pull this in once its cleaned up.

**Comment by mosabua** [comment]
Too old .. not going to worry unless this is reopened or replaced with a new PR.

**Pull request closed without merging**